### DISCUSSION TITLE
Create file in specific folder

### DISCUSSION THREAD

**Discussion by cdmcdermott** [SilentVoid13/Templater#1170] [Help] [answered]
I have several templates setup that create new files in the same folder. I would now like to have one template create the new file in a different folder. Sounds simple, but can't get it to work. I have tried to follow #649 but it seems to be a slightly different request as they would like to create a new file, then move, then add more information. 

Here is what I have so far, but returns a parsing error

<% let folder = tp.file.find_tfile("07 Planner/07.6 Exercise"); %>
<% let newFileName = tp.date.now("YYYY-MM-DD") + "_E.md"; %>
<% tp.file.create(tfile(folder.path + "/" + newFileName)); %>
Created: <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>

**Comment by Zachatoo** [comment]
1. You are defining variables outside of an [execution command](https://silentvoid13.github.io/Templater/commands/execution-command.html), which I'm not sure if that works or not.
1. `tp.file.find_tfile()` is for finding markdown files, not folders. You should use `app.vault.getAbstractFileByPath()` instead from the Obsidian API.
1. The function to create a new file is `tp.file.create_new()`, see the [docs](<https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilecreate_newtemplate-tfile--string-filename-string-open_new-boolean--false-folder-tfolder>) for more information on how to use it.

```js
<%*
let folder = app.vault.getAbstractFileByPath("07 Planner/07.6 Exercise");
let newFileName = tp.date.now("YYYY-MM-DD") + "_E.md";
tp.file.create_new("", newFileName, false, folder);
%>
Created: <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
```

In the future when asking for help, please include any error messages you get. You can view the full error messages in the console, which you can open with `cmd shift i`.

**Reply by cdmcdermott** [reply]
Thanks @Zachatoo 

tp.file.find_tfile() - crikey missed that.

Ok thanks for the solution, however, it doesn't quite work. In creates two files:
 - one file with correct content but called 'Untitled' in the default templater folder. 
 - second blank file, but with correct name and folder location

No error messages

**Reply by Zachatoo** [reply]
That is correct, what were you expecting to happen? I must've misunderstood what you were trying to do.

If you want content to go in the newly created file, you need to pass it content in the first argument instead of an empty string.

**Comment by cdmcdermott** [comment]
Apologies, I probably did not make myself clear. I am trying to create one file called 'YYYY-MM-DD_E' in the folder '07 Planner/07.6 Exercise' with the specified frontmatter AND also the body content of the template.

Your code creates file 'YYYY-MM-DD_E' in folder '07 Planner/07.6 Exercise' but is completely blank. The 'untitled' file has correct frontmatter and content body, but has wrong name and location.

I originally tried creating the file and then moving it `tp.file.move`

**Reply by Zachatoo** [reply, marked as answer]
That's a lot more straightforward than I thought, you can just move and rename the file, and setup the file how you want it.

This is assuming you're using the "create new file from template" command, which I'm assuming since in your other message you said you ended up with two files.

```js
<%*
let newFileName = tp.date.now("YYYY-MM-DD") + "_E.md";
await tp.file.move("07 Planner/07.6 Exercise/" + newFileName);
-%>
<% "---" %>
Created: <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
<% "---" %>
Created: <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
```

**Reply by cdmcdermott** [reply]
Thanks. This was actually similar to how I initially approached it before getting side tracked after looking at a similar question. I was missing the `await`  keyword to pause execution of the function. 
Solution accepted

**Comment by luccomo** [comment]
thanks for all the insights all.
I am trying to find a solution to the following
this code is in my "Daily Template" which creates a Daily Note,  the code below is to also create my Work Daily note in a folder. It works but I would like to also have the link to the WorkDaily note in my Daily note
`let folder = app.vault.getAbstractFileByPath("1 - WorkDaily");
let newFileName = tp.date.now("YYYY-MM-DD") + "_WorkDaily.md";
tp.file.create_new("#tag1 #WorkDailyTag", newFileName, false, folder)
%>`
Anyone know how to add to code a return of the newFileName as a link in the Daily Note
thanks all

**Reply by Zachatoo** [reply]
Sure, just wrap the filename in `[[]]` later on in your template

```
[[<% newFileName %>]]
```

**Reply by luccomo** [reply]
Awesome thanks @Zachatoo
